File: 0014-longest-common-prefix/0014-longest-common-prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        StringBuilder prefix = new StringBuilder();
        int minLength = int.MaxValue;
        
        foreach (var s in strs) {
            minLength = Math.Min(minLength, s.Length);
        }
        
        for (int i = 0; i < minLength; i++) {
            char c = strs[0][i];
            foreach (string str in strs) {
                if (str[i] != c) return prefix.ToString();
            }
            prefix.Append(c);
        }
        
        return prefix.ToString();
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
const longestCommonPrefix = function(strs) {
    let prefix = '';
    for (let i = 0; i < strs[0].length; i++) {
        const char = strs[0][i];
        for (const words of strs) {
            if (words[i] !== char) {
                return prefix;
            }
        }
        prefix += char;
    }
    return prefix;
};
